src/js/components/award/details/additional/AssistanceAdditionalDetails.jsx

```diff
@@ -14,12 +17,95 @@ const propTypes = {
 };
 
 export default class AssistanceAdditionalDetails extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hasLoaded: false,
+            inFlight: false,
+            items: {}
+        };
+    }
+    componentDidMount() {
+        this.loadAdditionalInfo();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.award.selectedAward.id !== prevProps.award.selectedAward.id) {
+            this.loadAdditionalInfo();
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.financialRequest) {
+            this.financialRequest.cancel();
+        }
+    }
+
+
+    loadAdditionalInfo() {
+        const awardId = this.props.award.selectedAward.id;
+        const grantFields = [
+            'treasury_account',
+            'object_class',
+            'program_activity'
+        ];
+        if (!awardId) {
+            return;
+        }
+
+        this.setState({
+            inFlight: true
+        });
+
+        if (this.financialRequest) {
+            this.financialRequest.cancel();
+        }
+
+        this.financialRequest = SearchHelper.performFinancialSystemLookup({
+            verbose: true,
+            filters: [
+                {
+                    field: 'award',
+                    operation: 'equals',
+                    value: awardId
+                }
+            ],
+            order_by: 'reporting_period_start',
+            limit: 1,
+            fields: grantFields
+        });
+
+        this.financialRequest.promise
+            .then((res) => {
+                let finItem = {};
+
+                res.data.results.forEach((item) => {
+                    finItem = new FinancialSystemItem(item);
+                });
+
+
+                this.setState({
+                    hasLoaded: true,
+                    inFlight: false,
```

Comment:
Can we use the `inFlight` state to prevent the group values from display "Not available" while the API request is still pending? Right now it flickers to "Not available" until the API responds.